src/displayapp/screens/settings/SettingWakeUp.cpp

```diff
@@ -89,19 +89,51 @@ bool SettingWakeUp::Refresh() {
 
 
 void SettingWakeUp::UpdateSelected(lv_obj_t *object, lv_event_t event) {
-  if(event == LV_EVENT_VALUE_CHANGED) {
+  if(event == LV_EVENT_VALUE_CHANGED && !isLocked) {
+    isLocked = true;
+    bool anyCheckboxChecked = false;
```

Comment:
Setting anyCheckboxChecked to false here does not make much sense since you will have to check the status of other buttons.

Result now:
- Click single tap
- Click wake on wrist
- click single tap again
- This removes wake on wrist also and sets it to None

It makes more sense to have it as:
`bool anyCheckboxChecked = !lv_checkbox_is_checked(cbOption[0]);`

Reply:
Can you try the new commit?

Reply:
Works :)

Reply:
Great!